tests/unit/test_instance.py

```diff
@@ -14,17 +14,52 @@
 
 import asyncio
 from datetime import datetime, timedelta
+from typing import Tuple
 
 import aiohttp
 from mocks import FakeAlloyDBClient
 import pytest
 
 from google.cloud.alloydb.connector.exceptions import RefreshError
-from google.cloud.alloydb.connector.instance import Instance
+from google.cloud.alloydb.connector.instance import _parse_instance_uri, Instance
 from google.cloud.alloydb.connector.refresh import _is_valid, RefreshResult
 from google.cloud.alloydb.connector.utils import generate_keys
 
 
+@pytest.mark.parametrize(
+    "instance_uri, expected",
+    [
+        (
+            "projects/test-project/locations/test-region/clusters/test-cluster/instances/test-instance",
+            ("test-project", "test-region", "test-cluster", "test-instance"),
+        ),
+        (
+            "projects/test-domain:test-project/locations/test-region/clusters/test-cluster/instances/test-instance",
+            (
+                "test-domain:test-project",
+                "test-region",
+                "test-cluster",
+                "test-instance",
+            ),
+        ),
+    ],
+)
+def test_parse_instance_uri(instance_uri: str, expected: Tuple[str, str, str]) -> None:
```

Comment:
I think we need the 4th `str` for the tuple...
```suggestion
def test_parse_instance_uri(instance_uri: str, expected: Tuple[str, str, str, str]) -> None:
```

Reply:
my bad, I have added and pushed again with latest changes.